packages/ui/src/working-groups/modals/CreateOpening/CreateOpening.tsx

```diff
@@ -0,0 +1,176 @@
+import BN from 'bn.js'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import styled from 'styled-components'
+
+import { useTransactionFee } from '@/accounts/hooks/useTransactionFee'
+import { useApi } from '@/api/hooks/useApi'
+import { ButtonPrimary, ButtonsGroup } from '@/common/components/buttons'
+import { DownloadButtonGhost } from '@/common/components/buttons/DownloadButtons'
+import { Modal, ModalHeader, ModalTransactionFooter } from '@/common/components/Modal'
+import {
+  /*StepDescriptionColumn, */ Stepper,
+  StepperModalBody,
+  StepperModalWrapper,
+} from '@/common/components/StepperModal'
+import { TextMedium } from '@/common/components/typography'
+import { useMachine } from '@/common/hooks/useMachine'
+import { useModal } from '@/common/hooks/useModal'
+import { SignTransactionModal } from '@/common/modals/SignTransactionModal/SignTransactionModal'
+import { isLastStepActive } from '@/common/modals/utils'
+import { getSteps } from '@/common/model/machines/getSteps'
+import { useYupValidationResolver } from '@/common/utils/validation'
+import { machineStateConverter } from '@/council/modals/AnnounceCandidacy/helpers'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+import { StyledStepperBody } from '@/proposals/modals/AddNewProposal'
+import { GroupIdToGroupParam } from '@/working-groups/constants'
+
+import { SuccessModal, CreateOpeningSteps as Steps, ImportOpening } from './components'
+import { createOpeningMachine, CreateOpeningMachineState, getTxParams } from './machine'
+import { OpeningConditions, CreateOpeningForm, CreateOpeningModalCall, OpeningSchema, defaultValues } from './types'
+
+export const CreateOpeningModal = () => {
+  const [showImport, setShowImport] = useState<boolean>(false)
+
+  const { api, connectionState } = useApi()
+  const { active: activeMember } = useMyMemberships()
+  const [state, send, service] = useMachine(createOpeningMachine)
+  const { hideModal, modalData } = useModal<CreateOpeningModalCall>()
+
+  const { group } = modalData
+  const workingGroupConsts = api?.consts[group]
+
+  const context = {
+    hiringTarget: 1,
```

Comment:
Unlike for the create lead opening proposal here the hiring target won't necessary be 1. E.g this opening has a hiring target of 2: https://pioneerapp.xyz/#/working-groups/openings/builders-32

I think this field should be added to `DurationAndProcess` (here there probably should be a toggle to have this field on the create opening modal but not have it on the create proposal modal).